gpu4pyscf/df/df.py

```diff
@@ -198,103 +199,164 @@ def reset(self, mol=None):
     get_ao_eri = get_eri = NotImplemented
     get_mo_eri = ao2mo = NotImplemented
 
-def cholesky_eri_gpu(intopt, mol, auxmol, cd_low, omega=None, sr_only=False):
+def cholesky_eri_gpu(intopt, mol, auxmol, cd_low, 
+                     omega=None, sr_only=False, use_gpu_memory=True):
     '''
     Returns:
         2D array of (naux,nao*(nao+1)/2) in C-contiguous
     '''
-    naoaux, naux = cd_low.shape
-    npair = len(intopt.cderi_row)
+    naux = cd_low.shape[1]
+    npairs = len(intopt.cderi_row)
     log = logger.new_logger(mol, mol.verbose)
-    nq = len(intopt.log_qs)
+    num_gpus = cupy.cuda.runtime.getDeviceCount()
 
     # if the matrix exceeds the limit, store CDERI in CPU memory
+    # TODO: better estimate of memory consumption for each device
     avail_mem = get_avail_mem()
-    use_gpu_memory = True
-    if naux * npair * 8 < 0.4 * avail_mem:
-        try:
-            cderi = cupy.empty([naux, npair], order='C')
-            log.debug(f"Saving CDERI on GPU. CDERI size {cderi.nbytes/GB}")
-        except Exception:
-            use_gpu_memory = False
+    
+    if use_gpu_memory:
```

Comment:
Should the `num_gpus` be considered when estimating GPU memory capacity?

Reply:
Yes, there is a todo in the comment. Will do in next PR.